TcgPocket/App/src/pages/settings/modules/PasswordForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { useForm } from '@mantine/form';
+import { useFormValidation } from '../../../helpers/useFormValidation';
+import {
+  SignInUserDto,
+  UserGetDto,
+  UserPasswordUpdateDto,
+} from '../../../types/users';
+import { Flex, PasswordInput, createStyles } from '@mantine/core';
+import { SecondaryButton } from '../../../components/buttons/SecondaryButton';
+import { PrimaryButton } from '../../../components/buttons/PrimaryButton';
+import { dispatch } from '../../../store/configureStore';
+import { updateUserPassword } from '../../../services/AuthServices';
+import { error, success } from '../../../services/notification';
+
+type PasswordFormProps = {
+  user: UserGetDto;
+  loginUserAfterUpdate: (arg: SignInUserDto) => void;
+};
+
+type PasswordForm = Omit<UserPasswordUpdateDto, 'userName'>;
+
+const initialValues = {
+  currentPassword: '',
+  newPassword: '',
+  newPasswordConfirmation: '',
+} as const;
+
+export function PasswordForm({
+  user,
+  loginUserAfterUpdate,
+}: PasswordFormProps): React.ReactElement {
+  const { classes } = useStyles();
+  const { validatePassword } = useFormValidation();
+
+  const form = useForm({
+    initialValues: initialValues,
+    validateInputOnBlur: true,
+    validate: {
+      currentPassword: (value) => (value === '' ? 'Cannot be Empty' : null),
+      newPassword: (value) =>
+        validatePassword(value) ? 'Invalid Password' : null,
+      newPasswordConfirmation: (value, values) =>
+        value !== values.newPassword ? 'Passwords do not match' : null,
```

Comment:
Is there any way we can check to make sure we arent making a useless call if someone tries to update their password with the same password they already have? This could be something we could handle on the backend but I think it would be nice to reduce API call. Would like to know your thoughts on this.

Reply:
see <conversation we had before you left>

Reply:
![image](https://github.com/gbrad777/TcgPocket/assets/113717324/912c6657-6c68-4a47-bf0d-26d2aca61be2)
```suggestion
      currentPassword: (value) => (value === '' ? 'Cannot be Empty' : null),
       newPassword: (value, values) =>
          value === values.currentPassword
          ? 'Must be different than current password.'
          : validatePassword(value)
          ? 'Invalid Password'
          : null,
```
This works to do what I was trying to describe.